dissect/ntfs/mft.py

```diff
@@ -120,12 +120,15 @@ def get(self, ref: Union[int, str, Instance], root: Optional[MftRecord] = None)
         else:
             raise TypeError(f"Invalid MFT reference: {ref!r}")
 
-    def segments(self) -> Iterator[MftRecord]:
+    def segments(self, start: int = 0, end: int = -1) -> Iterator[MftRecord]:
         """Yield all valid MFT records, regardless if they're allocated or not."""
         record_size = self.ntfs._record_size if self.ntfs else DEFAULT_RECORD_SIZE
-        mft_size = self.get(FILE_NUMBER_MFT).size()
+        end = (self.get(FILE_NUMBER_MFT).size() // record_size) if end == -1 else end
 
-        for segment in range(mft_size // record_size):
+        if start > end:
+            raise Error("The start segment cannot be higher than the end segment number")
```

Comment:
On second thought, why shouldn't this be allowed? Maybe nice to allow this so that you can go backwards as well.

You'd need to add the same logic for handling `-1` for `start` as you currently do for `end` though. And set the step for `range` to `-1`:

```
step = 1 if start <= end else -1
for segment in range(start, end + step, step):
```